### PR TITLE
chromium: Update to 122.0.6261.128

### LINKED ISSUE #795: Update to Chromium 122
Release notes:
* https://chromereleases.googleblog.com/2024/02/stable-channel-update-for-desktop_20.html
* https://chromereleases.googleblog.com/2024/02/stable-channel-update-for-desktop_22.html
* https://chromereleases.googleblog.com/2024/02/stable-channel-update-for-desktop_27.html
* https://chromereleases.googleblog.com/2024/03/stable-channel-update-for-desktop.html
* https://chromereleases.googleblog.com/2024/03/stable-channel-update-for-desktop_12.html

### PR DESCRIPTION
**Pull request by MaxIhlenfeldt**
Fixes #795.

Release notes:
    https://chromereleases.googleblog.com/2024/02/stable-channel-update-for-desktop_20.html
    https://chromereleases.googleblog.com/2024/02/stable-channel-update-for-desktop_22.html
    https://chromereleases.googleblog.com/2024/02/stable-channel-update-for-desktop_27.html
    https://chromereleases.googleblog.com/2024/03/stable-channel-update-for-desktop.html
    https://chromereleases.googleblog.com/2024/03/stable-channel-update-for-desktop_12.html

Build and patch changes:
------------------------

Drop six backported patches that are included upstream now.

Drop arm/0002-Fix-static_assert-failed-build-error-on-32-bit-ARM.patch as it was only needed to fix a build error with clang < 17, which we don't support any more.

Rebase remaining patches and renumber remaining ARM patches.

License changes:
----------------

Added licenses:
  * third_party/coremltools/mlmodel/format/LICENSE
  - https://chromium-review.googlesource.com/c/chromium/src/+/5108678

  * third_party/node/LICENSE
  - https://chromium-review.googlesource.com/c/chromium/src/+/5136956

  * third_party/angle/src/libANGLE/renderer/vulkan/shaders/src/third_party/etc_decoder/LICENSE, third_party/angle/src/third_party/libXNVCtrl/LICENSE, third_party/angle/src/third_party/volk/LICENSE.md, third_party/angle/third_party/android_system_sdk/LICENSE, third_party/angle/third_party/flatbuffers/LICENSE, third_party/catapult/experimental/trace_on_tap/third_party/pako/LICENSE, third_party/catapult/third_party/Paste/docs/license.txt, third_party/catapult/third_party/WebOb/docs/license.txt, third_party/catapult/third_party/apiclient/LICENSE, third_party/catapult/third_party/cachetools/LICENSE, third_party/catapult/third_party/chardet/LICENSE, third_party/catapult/third_party/cloudstorage/COPYING, third_party/catapult/third_party/coverage/LICENSE.txt, third_party/catapult/third_party/flot/LICENSE.txt, third_party/catapult/third_party/google-auth/LICENSE, third_party/catapult/third_party/gsutil/LICENSE, third_party/catapult/third_party/html5lib-python/LICENSE, third_party/catapult/third_party/httplib2/LICENSE, third_party/catapult/third_party/ijson/LICENSE.txt, third_party/catapult/third_party/oauth2client/LICENSE, third_party/catapult/third_party/polymer/LICENSE.polymer, third_party/catapult/third_party/polymer2/LICENSE.polymer, third_party/catapult/third_party/polymer3/LICENSE.polymer, third_party/catapult/third_party/pyfakefs/COPYING, third_party/catapult/third_party/pyparsing/LICENSE, third_party/catapult/third_party/python_gflags/COPYING, third_party/catapult/third_party/redux/LICENSE.md, third_party/catapult/third_party/requests_toolbelt/LICENSE, third_party/catapult/third_party/six/LICENSE, third_party/catapult/third_party/typ/LICENSE, third_party/catapult/third_party/vinn/third_party/v8/LICENSE, third_party/catapult/third_party/vinn/third_party/v8/LICENSE.strongtalk, third_party/catapult/third_party/vinn/third_party/v8/LICENSE.v8, third_party/catapult/third_party/vinn/third_party/v8/LICENSE.valgrind, third_party/catapult/third_party/webapp2/LICENSE, third_party/catapult/tracing/third_party/jpeg-js/LICENSE, third_party/catapult/tracing/third_party/oboe/LICENCE, third_party/catapult/tracing/third_party/pako/LICENSE, third_party/chromevox/third_party/closure-library/LICENSE, third_party/chromevox/third_party/sre/LICENSE, third_party/crashpad/crashpad/third_party/getopt/LICENSE, third_party/crashpad/crashpad/third_party/xnu/APPLE_LICENSE, third_party/dawn/third_party/khronos/LICENSE, third_party/devtools-frontend/src/front_end/third_party/axe-core/LICENSE, third_party/devtools-frontend/src/front_end/third_party/lodash-isequal/LICENSE, third_party/devtools-frontend/src/front_end/third_party/vscode.web-custom-data/package/LICENSE.md, third_party/google_input_tools/third_party/closure_library/LICENSE, third_party/openscreen/src/third_party/protobuf/LICENSE, third_party/openscreen/src/third_party/tinycbor/src/LICENSE, third_party/openscreen/src/third_party/zlib/LICENSE, third_party/pdfium/third_party/fp16/LICENSE, third_party/pdfium/third_party/freetype/FTL.TXT, third_party/screen-ai/LICENSE, third_party/selenium-atoms/LICENSE.closure, third_party/selenium-atoms/LICENSE.sizzle, third_party/selenium-atoms/LICENSE.wgxpath, third_party/swiftshader/third_party/SPIRV-Headers/LICENSE, third_party/swiftshader/third_party/SPIRV-Tools/LICENSE, third_party/webrtc/common_audio/third_party/ooura/LICENSE, third_party/webrtc/common_audio/third_party/spl_sqrt_floor/LICENSE, third_party/webrtc/modules/third_party/portaudio/LICENSE, third_party/webrtc/rtc_base/third_party/sigslot/LICENSE,
  - https://chromium-review.googlesource.com/c/chromium/src/+/5061468

Removed licenses:
  * third_party/gvr-android-sdk/LICENSE
  - https://chromium-review.googlesource.com/c/chromium/src/+/5128757

  * third_party/nasm/LICENSE
  - Removed due to being marked as not shipped.

Updated licenses:
  * chrome/installer/mac/third_party/xz/LICENSE
  - https://chromium-review.googlesource.com/c/chromium/src/+/5289735

  * third_party/libxml/src/Copyright
  - https://chromium-review.googlesource.com/c/chromium/src/+/5140988

Test-built (big thanks to @rjanani-p):
--------------------------------------

* chromium-ozone-wayland:
 - master, clang,   MACHINE=qemux86-64, raspberrypi4-64
 - nanbield, clang, MACHINE=qemuarm64, qemux86-64

* chromium-x11:
 - master, clang,   MACHINE=qemuarm, qemuarm64, qemux86-64
 - nanbield, clang, MACHINE=qemuarm, qemuarm64, qemux86-64, raspberrypi4-64

Signed-off-by: Max Ihlenfeldt <max@igalia.com>

### CODE REVIEW

**Comment by MaxIhlenfeldt** [comment]
Turns out only around 70 licenses are now included due to upstream changes to the license script. The previous and much higher number I reported was due to me messing up the script so that non-shipped licenses were included as well :sweat_smile: Fortunately, these added licenses are all located under easily recognizable paths, so I think I managed to spot all licenses that were actually new.

@rjanani-p as always, testing would be greatly appreciated!

**Comment by MaxIhlenfeldt** [comment]
Also, I can't remember how long it's been since I was able to upload a PR without adding any new patches... not having to support clang < 17 is great!

File: meta-chromium/recipes-browser/chromium/generate_license_list.py [hunk 1]
```diff
@@ -60,17 +60,28 @@ def find_chromium_licenses(chromium_root):
...
+        errors = []
         try:
-            metadata_list = licenses.ParseDir(d, chromium_root)
+            metadata_list, errors = licenses.ParseDir(d, chromium_root)
         except licenses.LicenseError as e:
-            print('Exception in directory %s: %s' % (d, e))
+            errors.append(str(e))
+
+        if len(errors) != 0:
+            # Recent changes to the upstream script have resulted in a huge mass of
```
[the remark this answers is not in the document]
Comment (MaxIhlenfeldt) [reply meta-chromium/recipes-browser/chromium/generate_license_list.py]: Done

File: meta-chromium/recipes-browser/chromium/generate_license_list.py [hunk 2]
```diff
@@ -60,17 +60,28 @@ def find_chromium_licenses(chromium_root):
...
+            # errors. Going through all of them isn't feasible, so we'll have to
+            # ignore them for now.
+            pass
+
+            '''
```
[the remark this answers is not in the document]
Comment (MaxIhlenfeldt) [reply meta-chromium/recipes-browser/chromium/generate_license_list.py]: Done

**Review by rakuco — APPROVED** [review]
lgtm! Nice to see so many patches going away!

**Comment by rjanani-p** [comment]
> Turns out only around 70 licenses are now included due to upstream changes to the license script. The previous and much higher number I reported was due to me messing up the script so that non-shipped licenses were included as well 😅 Fortunately, these added licenses are all located under easily recognizable paths, so I think I managed to spot all licenses that were actually new.
> 
> @rjanani-p as always, testing would be greatly appreciated!

@MaxIhlenfeldt ,I have completed the testing for Poky master with latest changes, sharing the results for reference:

1.**For chromium-x11 & qemux86-64 target:** compiled successfully without build issues, able to check chromium version using qemux86-64 emulator

2.**For chromium-x11 & qemuarm target:** compiled successfully without build issues, to check chromium version using qemuarm emulator unable to check it due to curser / mouse is not working properly while operating with chromium browser

3.**For chromium-x11 & qemuarm64 target:** compiled successfully without build issues, able to check chromium version using qemuarm64 emulator

4.**For chromium-ozone-wayland & qemux86-64 target**: compiled successfully without build issues, able to check chromium version using qemux86-64 emulator
5. *For chromium-ozone-wayland & raspberrypi4-64 target**: compiled successfully without build issues, didn't check with hardware due to hardware unavailability.

Next, I will start testing for poky nanbield.

**Comment by rjanani-p** [comment]
> Turns out only around 70 licenses are now included due to upstream changes to the license script. The previous and much higher number I reported was due to me messing up the script so that non-shipped licenses were included as well 😅 Fortunately, these added licenses are all located under easily recognizable paths, so I think I managed to spot all licenses that were actually new.
> 
> @rjanani-p as always, testing would be greatly appreciated!

@MaxIhlenfeldt , getting compilation issue for poky nanbied & target qemux86-64 for chromium-x11 recipe

| [268/58024] python3 ../../third_party/node/node.py ../../third_party/devtools-frontend/src/front_end/core/i18n/collect-ui-strings.js --input-directories ../../third_party/devtools-frontend/src/front_end --output-directory gen/third_party/devtools-frontend/src/front_end/core/i18n/locales/generated
| FAILED: gen/third_party/devtools-frontend/src/front_end/core/i18n/locales/generated/en-US.json
| python3 ../../third_party/node/node.py ../../third_party/devtools-frontend/src/front_end/core/i18n/collect-ui-strings.js --input-directories ../../third_party/devtools-frontend/src/front_end --output-directory gen/third_party/devtools-frontend/src/front_end/core/i18n/locales/generated
| Traceback (most recent call last):
|   File "/build_chrom_122_poky_nanbield/tmp/work/core2-64-poky-linux/chromium-x11/122.0.6261.128/chromium-122.0.6261.128/out/Release/../../third_party/node/node.py", line 39, in <module>
|     RunNode(sys.argv[1:])
|   File "/build_chrom_122_poky_nanbield/tmp/work/core2-64-poky-linux/chromium-x11/122.0.6261.128/chromium-122.0.6261.128/out/Release/../../third_party/node/node.py", line 34, in RunNode
|     raise RuntimeError('Command \'%s\' failed\n%s' % (' '.join(cmd), err))
| RuntimeError: Command '/build_chrom_122_poky_nanbield/tmp/work/core2-64-poky-linux/chromium-x11/122.0.6261.128/chromium-122.0.6261.128/out/Release/../../third_party/node/linux/node-linux-x64/bin/node ../../third_party/devtools-frontend/src/front_end/core/i18n/collect-ui-strings.js --input-directories ../../third_party/devtools-frontend/src/front_end --output-directory gen/third_party/devtools-frontend/src/front_end/core/i18n/locales/generated' failed
| /build_chrom_122_poky_nanbield/tmp/work/core2-64-poky-linux/chromium-x11/122.0.6261.128/chromium-122.0.6261.128/third_party/devtools-frontend/src/node_modules/glob/common.js:19
|   return a.localeCompare(b, 'en')
|            ^
|
| RangeError: Internal error. Icu error.
|     at String.localeCompare (<anonymous>)

**Comment by MaxIhlenfeldt** [comment]
> Error is expected for poky nanbield ?

No, and I don't see it in my local nanbield build for qemux86-64. What's the output when you run `/build_chrom_122_poky_nanbield/tmp/work/core2-64-poky-linux/chromium-x11/122.0.6261.128/chromium-122.0.6261.128/third_party/node/linux/node-linux-x64/bin/node --version`?

**Comment by rjanani-p** [comment]
> > Error is expected for poky nanbield ?
> 
> No, and I don't see it in my local nanbield build for qemux86-64. What's the output when you run `/build_chrom_122_poky_nanbield/tmp/work/core2-64-poky-linux/chromium-x11/122.0.6261.128/chromium-122.0.6261.128/third_party/node/linux/node-linux-x64/bin/node --version`?

/chromium_ver_122_builds/builds_chromium/build_chrom_122_poky_nanbield/tmp/work/core2-64-poky-linux/chromium-x11/122.0.6261.128/chromium-122.0.6261.128/third_party/node/linux/node-linux-x64/bin/node --version
**v20.8.1**

getting version as **v20.8.1**, for 3 targets (qemux86-64, qemuarm, qemuarm64) getting same build issue

**Comment by MaxIhlenfeldt** [comment]
Hm, that's the same version I see. I'm not sure what could be causing this, do you maybe have a `repo` manifest you can share with me, together with all your local `.conf` files, so I can replicate your setup as close as possible?

**Comment by rjanani-p** [comment]
@MaxIhlenfeldt ,I have cross checked the Poky nanbield with latest changes and clean build, didn't face any compilation issues and completed the testing below configuration. Here I am sharing the results for reference:

1.For chromium-x11 & qemux86-64 target: compiled successfully without build issues, able to check chromium version using qemux86-64 emulator

2.For chromium-x11 & qemuarm target: compiled successfully without build issues, to check chromium version using qemuarm emulator unable to check it due to curser / mouse is not working properly while operating with chromium browser

3.For chromium-x11 & qemuarm64 target: compiled successfully without build issues, able to check chromium version using qemuarm64 emulator

4. For chromium-x11 & raspberrypi4-64 target: compiled successfully without build issues, didn't check with hardware due to hardware unavailability.

5.For chromium-ozone-wayland & qemux86-64 target: compiled successfully without build issues, able to check chromium version using qemux86-64 emulator

**Comment by MaxIhlenfeldt** [comment]
@rjanani-p iiuc this means there are no issues and we can go ahead and merge this?

**Comment by rjanani-p** [comment]
> @rjanani-p iiuc this means there are no issues and we can go ahead and merge this?

yes @MaxIhlenfeldt we can merge, during my testing didn't face issues.

**Review by Gdakgdak22 — APPROVED** [review]

**Comment by MaxIhlenfeldt** [comment]
Updated the Test-built section in the commit message and PR description, we're good to go!

**Pull request merged**